### PR TITLE
Feat/hype creation

### PR DESCRIPTION
**Pull request by Malak67**
# Generic Pull Request template

## Linked issue

Please link the issue:

ex. Closes #<33>.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.
- [x] The changes impact front-end.

## For changes impacting frontend

Please leave some screenshots/video recordings of the newly implemented interface and the Figma link for its initial design.

### Screenshots/ recordings

=> Multimedia here.

### Figma link

=> Figma link here.

## Description

Please provide a short description about what the PR contains.

## Funny gif

Leave a funny gif for faster review.

### CODE REVIEW

File: services/client/src/constants/addresses.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export const hypeAddress: string =
-  `${process.env.REACT_APP_HYPE_ADDRESS}` || '0x869b1169DD8dB7A143aDc8190C28Ca4799aa152c';
+  `${process.env.REACT_APP_HYPE_ADDRESS}` || '0x569e9b6580B9C24148dc8107C2F5281F30Fff7fa';
```
Comment (VargaElod23) [inline services/client/src/constants/addresses.ts:2]: I don't think we need these defaults here

Comment (VargaElod23) [reply services/client/src/constants/addresses.ts:2]: Its better to error out than use the wrong contract

File: services/client/src/constants/ipfs.ts [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
-  host: `ipfs.infura.io`,
-  port: 5001,
-  protocol: `https`,
-  headers: {
+const ipfsHost = `${process.env.REACT_APP_IPFS_HOST}`;
+const ipfsPort = +process.env.REACT_APP_IPFS_PORT;
+const ipfsProtocol = `${process.env.REACT_APP_IPFS_PROTOCOL}`;
+const ipfsUseAuth = `${process.env.REACT_APP_IPFS_USE_AUTH}`;
+
+// export const ipfsClient = create({
```
Comment (VargaElod23) [inline services/client/src/constants/ipfs.ts:15]: Can you add a @dev note to mark this as a local dev setup and move it to the end of the file

File: services/client/src/hooks/useContractActivatePool.tsx [hunk 3]
```diff
@@ -38,12 +40,11 @@ const useContractActivatePool = (id: number) => {
     },
   });
 
-  const waitForTransaction = useWaitForTransaction({
+  useWaitForTransaction({
     hash: data?.hash,
-    // wait: poolData?.wait,
     onSuccess(transactionData) {
+      console.log('Successfully called activate pool', transactionData);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractActivatePool.tsx:46]: 🧹

File: services/client/src/hooks/useContractActivatePool.tsx [hunk 4]
```diff
@@ -53,14 +54,30 @@ const useContractActivatePool = (id: number) => {
     onSettled(data, error) {
       console.log('Settled', { data, error });
       hideLoadingModal();
+      // const hypeI = new ethers.utils.Interface(abi);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractActivatePool.tsx:57]: 🧹

File: services/client/src/hooks/useContractActivatePool.tsx [hunk 5]
```diff
@@ -53,14 +54,30 @@ const useContractActivatePool = (id: number) => {
...
     },
   });
 
+  useEffect(() => {
+    if (enabled && id && typeof write === 'function') {
+      console.log('ID: ', id);
+      write();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [enabled, id]);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractActivatePool.tsx:76]: Here if you add `write` too it loops?

File: services/client/src/hooks/useContractCreatePool.tsx [hunk 6]
```diff
@@ -55,69 +57,60 @@ const useContractCreatePool = (
...
     onError(error: any) {
-      console.log('Error', error);
+      console.log('onError', error);
       hideLoadingModal();
       showNotificationModal(NotificationType.ERROR, error?.message);
       resetWriteContract();
     },
     onSettled(data, error) {
-      console.log('Settled', { data, error });
+      console.log('onSettled', { data, error });
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractCreatePool.tsx:91]: In my understanding, at final we should move everything from under onSettled under onSuccess, right? As both error and success get settled

File: services/client/src/hooks/useContractPoolURI.tsx [hunk 7]
```diff
@@ -5,11 +5,12 @@ import { useContractRead } from 'wagmi';
 
 const useContractPoolURI = (tokenId: number) => {
   const { abi } = ABIs.contracts.HypePool;
-  const hypeInterface = new utils.Interface(abi);
+  // const hypeInterface = new utils.Interface(abi);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractPoolURI.tsx:8]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 8]
```diff
@@ -1,100 +1,90 @@
...
+    // word: 'testnet',
+    title: 'Dragon Ball',
+    projectName: 'Dragon Ball Super, Dragon Ball, DBS',
+    tokenName: 'DBS',
+    description: 'Dragon Ball super NFT marketplace',
+    projectDescription: 'Something nice about DBS',
+    word: 'testnet',
   });
+  const [poolReward, setPoolReward] = useState<HypePoolRewardForm>({
+    // network: '',
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:49]: We should wait until the chain is selected by the user and initialize the reward with that chain's native token details

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 9]
```diff
@@ -1,100 +1,90 @@
...
+    // token: null,
+    // tokenAddress: '',
+    // tokenName: '',
+    // minReward: null,
+    // impressionReward: null,
+    // cap: null,
+    // endDate: null
+    network: 843,
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:57]: Also, can we export this to a config

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 10]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
       return;
     }
-    const poolCap = data.cap;
-    const projectName = data.projectName;
-    const description = data.description;
-    const title = data.title;
-    const tokenAddress = data.token;
-    const minHypeReward = data.minReward;
-    const endDate = data.endDate?.getTime();
+    const cap = BigNumber.from(rewards.cap).mul(BigNumber.from(10).pow(rewards.tokenDecimals));
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:129]:
```suggestion
    const cap = BigNumber.from(rewards.cap).mul((BigNumber.from(10).pow(rewards.tokenDecimals)));
```

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 11]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
+    const minReward = BigNumber.from(rewards.minReward).mul(
+      BigNumber.from(10).pow(rewards.tokenDecimals),
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:131]:
```suggestion
      (BigNumber.from(10).pow(rewards.tokenDecimals)),
```

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 12]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
+    );
+    const impressionReward = BigNumber.from(rewards.impressionReward).mul(
+      BigNumber.from(10).pow(rewards.tokenDecimals),
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:134]:
```suggestion
      (BigNumber.from(10).pow(rewards.tokenDecimals)),
```

Comment (VargaElod23) [reply services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:134]: Better safe than sorry 😃

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 13]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
     setContractEnabled(true);
   };
 
-  useEffect(() => {
-    if (isSubmitSuccessful) {
-      reset();
-    }
-  }, [isSubmitSuccessful, reset]);
+  const onSubmitDetails = async (data: HypePoolDetailsForm) => {
+    console.log('HypePoolDetailsForm: ', data);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:153]: 🧹 \

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 14]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
+    setPoolDetails(data);
+    // await onUploadToIpfs(data);
+    setCurrentStep(2);
+  };
+
+  const onSubmitRewards = (data: HypePoolRewardForm) => {
+    console.log('HypePoolRewardForm: ', data);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:160]: 🧹 \

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 15]
```diff
@@ -127,47 +122,68 @@ export const useAddHypePoolEffects = () => {
...
+  };
+
+  const onBackFromRewards = () => {
+    setCurrentStep(1);
+  };
 
-  const onCancel = () => {
-    reset();
+  const onFinalize = () => {
+    console.log('Finalize');
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:171]: 🧹 \

File: services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+  const [tokenAddress, setTokenAddress] = useState<`0x${string}`>(null);
+  const { chain } = useNetwork();
+  const isEthNetwork = chain?.name === 'Ethereum';
+  const { data: ERC20tokenInfo } = useToken({
+    address: tokenAddress,
+    enabled: !!tokenAddress && isEthNetwork,
+  });
+  const { changeNetwork } = useSwitchNetwork();
+
+  const networkOptions = [
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx:43]: Can we move this to a config

File: services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx [hunk 17]
```diff
@@ -0,0 +1,235 @@
...
+      name: 'Taraxa Devnet',
+      value: 843,
+    },
+    {
+      name: 'Taraxa Testnet',
+      value: 842,
+    },
+  ];
+
+  const tokensOptions = [
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx:62]: Same

File: services/client/src/pages/AddHypePool/Summary/Summary.effects.tsx [hunk 18]
```diff
@@ -0,0 +1,66 @@
+import { BigNumber } from 'ethers';
+import { useEffect, useState } from 'react';
+import useContractActivatePool from '../../../hooks/useContractActivatePool';
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/Summary/Summary.effects.tsx:3]: Maybe an index export there would be good to not have these lines here

File: services/client/src/utils/enums.ts [hunk 19]
```diff
@@ -27,3 +27,7 @@ export enum NotificationType {
 export type ButtonVariant = 'primary' | 'secondary' | 'success' | 'danger';
 
 export const AUTHENTICATION_TOKEN = 'AUTHENTICATION_TOKEN';
+
+export const zeroAddress = '0x0000000000000000000000000000000000000000';
```
Comment (VargaElod23) [inline services/client/src/utils/enums.ts:31]: We have this in ethers too, but its fine

**Review by VargaElod23 — COMMENTED** [review]
Some suggestions, but I think most of them will be tackled anyways by your refactor. Nonetheless, they're there.

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx (hunk 15 above)
Comment (Malak67) [reply services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:173]: I'll keep this until the last part of the wizzard is finished (after Hype Pool page is done)

File: services/client/src/hooks/useContractCreatePool.tsx [hunk 20]
```diff
@@ -1,124 +1,106 @@
...
-      console.log('Successfully minted Hype Pool transactionData', transactionData);
+      // console.log('onSuccess', transactionData);
       hideLoadingModal();
-      showSuccessModal();
+      successCallback();
       resetWriteContract();
     },
     onError(error: any) {
-      console.log('Error', error);
+      console.log('onError', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractCreatePool.tsx:75]: 🧹

File: services/client/src/hooks/useContractCreatePool.tsx [hunk 21]
```diff
@@ -1,124 +1,106 @@
...
-      console.log('On mutate');
-      showLoading();
-    },
-    onSuccess(data: any) {
-      console.log('Successfully called', data);
+      showLoading(['Please, sign the message...', 'Creating your Hype Pool on-chain...']);
     },
     onError(error: any) {
-      console.log('On error: ', error);
+      console.log('onError: ', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractCreatePool.tsx:59]: 🧹

File: services/client/src/hooks/useContractERC20Approve.tsx [hunk 22]
```diff
@@ -1,77 +1,87 @@
...
         'Please, sign the message...',
         'You need to approve in order to deposit your funds',
       ]);
     },
-    onSuccess(data: any) {
-      console.log('Successfully called', data);
-    },
     onError(error: any) {
-      console.log('On error: ', error);
+      console.log('onError', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractERC20Approve.tsx:54]: 🧹

File: services/client/src/hooks/useContractERC20Approve.tsx [hunk 23]
```diff
@@ -1,77 +1,87 @@
...
     hash: data?.hash,
-    wait: data?.wait,
     onSuccess(transactionData) {
       hideLoadingModal();
-      deposit();
-      console.log('Success', transactionData);
     },
     onError(error: any) {
-      console.log('Error', error);
+      console.log('onError', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractERC20Approve.tsx:66]: 🧹

File: services/client/src/hooks/useContractEscrowDeposit.tsx [hunk 24]
```diff
@@ -37,41 +35,39 @@ const useContractEscrowDeposit = (
     onMutate() {
       showLoading(['Please, sign the message...', 'Depositing rewards...']);
     },
-    onSuccess(data: any) {
-      console.log('Successfully called', data);
-    },
     onError(error: any) {
-      console.log('On error: ', error);
+      console.log('onError', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractEscrowDeposit.tsx:39]: 🧹

File: services/client/src/hooks/useContractEscrowDeposit.tsx [hunk 25]
```diff
@@ -37,41 +35,39 @@ const useContractEscrowDeposit = (
...
       hideLoadingModal();
       showNotificationModal(NotificationType.ERROR, error?.message);
     },
   });
 
-  const waitForTransaction = useWaitForTransaction({
+  useWaitForTransaction({
     hash: data?.hash,
-    wait: data?.wait,
+    // wait: data?.wait,
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractEscrowDeposit.tsx:47]: 🧹

File: services/client/src/hooks/useContractEscrowDeposit.tsx [hunk 26]
```diff
@@ -37,41 +35,39 @@ const useContractEscrowDeposit = (
...
     onSuccess(transactionData) {
-      console.log('Success', transactionData);
       hideLoadingModal();
-      showNotificationModal(NotificationType.SUCCESS, 'Funds successfully deposited');
     },
     onError(error: any) {
-      console.log('Error', error);
+      console.log('onError', error);
```
Comment (VargaElod23) [inline services/client/src/hooks/useContractEscrowDeposit.tsx:52]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 27]
```diff
@@ -1,100 +1,93 @@
...
+  const [isCustomToken, setIsCustomToken] = useState<boolean>(false);
+  const [poolTransaction, setPoolTransaction] = useState<string>();
+  const [poolDetails, setPoolDetails] = useState<HypePoolDetailsForm>({
     title: '',
+    projectName: '',
+    tokenName: '',
     description: '',
+    projectDescription: '',
+    word: 'testnet',
+    // title: 'Dragon Ball',
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:42]: I think we can delete this test data

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 28]
```diff
@@ -1,100 +1,93 @@
...
-  const {
-    handleSubmit,
-    register,
-    reset,
-    control,
-    formState: { isSubmitSuccessful, errors },
-  } = useForm({
-    defaultValues,
-    resolver: yupResolver(validationSchema),
+    // network: 843,
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:59]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 29]
```diff
@@ -1,100 +1,93 @@
...
+    successCallback,
+    setCreatedPoolIndex,
+    setPoolTransaction,
+  );
+
+  const onUploadToIpfs = async (data: HypePoolDetailsForm) => {
     const url = await uploadToIpfs(data);
     // const url = 'https://hype.infura-ipfs.io/ipfs/QmTuh1p9a9qGRWZ1QgzSrHfQ84HLEQ7n41VgM2Rd3yusfm';
     // const url = 'QmTuh1p9a9qGRWZ1QgzSrHfQ84HLEQ7n41VgM2Rd3yusfm';
     console.log('URL after upload: ', url);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:83]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 30]
```diff
@@ -108,7 +101,12 @@ export const useAddHypePoolEffects = () => {
...
     try {
-      const uploaded = await ipfsClient.add(JSON.stringify({ description: data?.description }));
+      const uploaded = await ipfsClient.add(
+        JSON.stringify({
+          description: data?.description,
+          projectDescription: data?.projectDescription,
+        }),
+      );
       url = uploaded.path;
       console.log('uploaded: ', uploaded);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:111]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 31]
```diff
@@ -127,47 +125,61 @@ export const useAddHypePoolEffects = () => {
...
 
-  useEffect(() => {
-    if (isSubmitSuccessful) {
-      reset();
-    }
-  }, [isSubmitSuccessful, reset]);
+  const onSubmitDetails = async (data: HypePoolDetailsForm) => {
+    setPoolDetails(data);
+    await onUploadToIpfs(data);
+    // setCurrentStep(2);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:159]: 🧹

File: services/client/src/pages/AddHypePool/AddHypePool.effects.tsx [hunk 32]
```diff
@@ -127,47 +125,61 @@ export const useAddHypePoolEffects = () => {
...
+  };
+
+  const onSubmitRewards = (data: HypePoolRewardForm) => {
+    createPool(poolDetails, data);
+    // setCurrentStep(3);
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/AddHypePool.effects.tsx:164]: 🧹

File: services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx [hunk 33]
```diff
@@ -0,0 +1,190 @@
...
+export interface HypePoolRewardForm
+  extends Pick<
+    AddHypePool,
+    'network' | 'token' | 'minReward' | 'impressionReward' | 'cap' | 'endDate'
+  > {
+  tokenAddress: string;
+  tokenName: string;
+  tokenDecimals: number;
+
+  // Some examples of custom ERC20 Tokens:
```
Comment (VargaElod23) [inline services/client/src/pages/AddHypePool/RewardForm/RewardForm.effects.tsx:20]: 🧹

File: services/client/src/utils/networks.ts [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+    blockExplorerUrl: 'https://explorer.devnet.taraxa.io/',
+    nativeCurrency: {
+      name: 'TARA',
+      symbol: 'TARA',
+      decimals: 18,
+    },
+  },
+};
+
+export const findInNetwork = (networkId: number): boolean => {
```
Comment (VargaElod23) [inline services/client/src/utils/networks.ts:64]:
```suggestion
export const findNetworkById = (networkId: number): boolean => {
```

**Comment by Malak67** [comment]
> 

Yep, I left them so they will help us debug it. I will remove them before merging it to master.

**Pull request merged**